base_gtin/static/src/js/barcode_parser.js

```diff
@@ -0,0 +1,56 @@
+/* Copyright 2016 LasLabs Inc.
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
+
+odoo.define('base_gtin.BarcodeParser', function(require) {
+    'use strict';
+
+    var BarcodeParser = require('barcodes.BarcodeParser');
+    
+    BarcodeParser.include({
+        
+        /* It performs EAN-14 checksum operation
+         * :param barcode: Barcode string to validate, including check digit
+         * :return: ``int`` representing calculated checksum
+         **/
+        ean14_checksum: function(barcode) {
+            if (barcode.length !== 14) {
```

Comment:
This is reimplemented in js because core also reimplements. Using XMLRPC would disallow the use of this code in systems that are offline, such as POS.